src/lib/encoder.ts

```diff
@@ -74,23 +74,58 @@ const encodeDataSourceWithHash = (
   const verificationMethod = getVerificationMethod(
     verification?.method || UNKNOWN_VERIFICATION_METHOD,
   );
-  return (
-    (verificationMethod
-      ? keccak256(verificationMethod.name).slice(0, 10)
-      : padLeft(0, 8)) +
-    stripHexPrefix(verification ? verification.data : padLeft(0, 64)) +
-    stripHexPrefix(utf8ToHex(dataSource))
-  );
+  return [
+    '0x0000',
+    stripHexPrefix(
+      verificationMethod
+        ? padLeft(keccak256(verificationMethod.name).slice(0, 10), 8)
+        : '00000000',
+    ),
+    stripHexPrefix(
+      verification?.data
+        ? padLeft(verification.data.slice(2).length / 2, 4)
+        : '0000',
+    ),
+    stripHexPrefix(
+      verification?.data ? stripHexPrefix(verification?.data) : '',
+    ),
+    stripHexPrefix(utf8ToHex(dataSource)),
+  ].join('');
 };
 
 const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
+  if (value.slice(0, 6) === '0x0000') {
+    /*
+      0        1         2         3         4         5         6         7         8
+      12345678901234567890123456789012345678901234567890123456789012345678901234567890
+      0x0000 code
+            6f357c6a hash fn [6]
+                    0020 data len [14]
+                        820464ddfac1be...[18 + data len]
+                                                       [18 + data len]...696670733a2f2...[...rest]
+    */
+    const verificationMethodSig = `0x${value.slice(6, 14)}`;
+    const verificationMethod = getVerificationMethod(verificationMethodSig);
+    if (verificationMethod !== undefined) {
+      const encodedLength = `0x${value.slice(14, 18)}`; // Rest of data string after function hash
+      const dataLength = hexToNumber(encodedLength, false) as number;
+      const dataHash = `0x${value.slice(18, 18 + dataLength * 2)}`; // Get jsonHash 32 bytes
+      const dataSource = hexToUtf8('0x' + value.slice(18 + dataLength * 2)); // Get remainder as URI
+      return {
+        verification: {
+          method: verificationMethod.name,
```

Comment:
```suggestion
          verificationMethod?.name || UNKNOWN_VERIFICATION_METHOD,
```

Reply:
Can name not be defined on verificationMethod ?

Reply:
The verificationMethod could be undefined and therefore the url is not validated

Reply:
I was more thinking of a situation where : it is defined but doesnt have a name property

Reply:
But we kind of want it to be undefined if not defined

Reply:
The default should be the signature hash so that reading a VerificationURI can return a yet unsupported custom verification method.